quinn-proto/src/connection/mod.rs

```diff
@@ -1500,6 +1577,37 @@ impl Connection {
         Ok(())
     }
 
+    fn detect_spurious_loss(&mut self, ack: &frame::Ack, space: SpaceId) -> bool {
+        let lost_packets = &mut self.spaces[space].lost_packets;
+
+        if !lost_packets.is_empty() {
+            for range in ack.iter() {
+                // TODO: avoid allocation
+                let spurious_losses = lost_packets
+                    .range(range.clone())
+                    .map(|(pn, _info)| pn)
+                    .copied()
+                    .collect::<Vec<_>>();
+
+                for pn in spurious_losses {
+                    lost_packets.remove(&pn);
+                }
+            }
+
+            lost_packets.is_empty()
+        } else {
+            false
+        }
+    }
+
+    /// Drain lost packets that we think will reasonably never arrive
+    fn drain_lost_packets(&mut self, now: Instant, space: SpaceId) {
+        let two_pto = 2 * self.path.rtt.pto_base();
```

Comment:
We should explain why this value was chosen, even if that explanation is just "blindly cribbed from msquic". Best of all would be to run some experiments and see where returns start to diminish in one or more real-world conditions, of course, but we can leave that to future work.